hw/ip/gpio/doc/theory_of_operation.md

```diff
@@ -73,14 +73,22 @@ the output driver is unable to switch the pin or during the delay imposed
 if the noise filter is enabled).
 
 #### GPIO Inputs as HW Straps
-The GPIO controller also provides an additional optional feature to sample the GPIO input values as hardware configuration straps
-These input values are sampled only at initial cold boot reset deassertion and is not sampled thereafter.
-Thus any change in the input configuration information after power on reset deassertion will not be captured.
-A strap_en signal driven by the pwrmgr is used for this sampling mechanism. The strap_en pulse is a single cycle pulse of the io div4 clock
-It is expected to toggle only once at cold boot.
-
-The snapshot value of the GPIO inputs is available in the [`HW_STRAPS_DATA_IN`](../data/gpio.hjson#hw_straps_data_in) register for firmware read out
-Note that all 32 bits of input data to the GPIO controller are sampled in this mode regardless of the gpio output enable status
+
+The GPIO controller provides an optional feature to sample GPIO input values as hardware configuration straps.
+
+After each reset, on the first cycle when the `strap_en_i` signal is asserted, the GPIO detects its
+transition from low to high. 
```

Comment:
Three broken sentences from here "After each reset..."; "One clock cycle..."; "This mechanism samples..."